.travis.yml

```diff
@@ -2,25 +2,38 @@ language: csharp
 solution: JSIL.sln
 
 before_install:
-  - wget -P /tmp/avim -N https://ftp.mozilla.org/pub/mozilla.org/firefox/nightly/latest-mozilla-central/jsshell-linux-x86_64.zip
-  - unzip -d /tmp/js /tmp/avim/jsshell-linux-x86_64.zip
+  - wget -P /tmp/ -N https://ftp.mozilla.org/pub/mozilla.org/firefox/nightly/latest-mozilla-central/jsshell-linux-x86_64.zip
+  - unzip -d /tmp/js /tmp/jsshell-linux-x86_64.zip
   - export PATH=$PATH:/tmp/js
 
 install:
   - export NODEJS=$(dirname `nvm which $NODEJS_VERSION`)
   - export FSHARPINSTALLDIR=$PWD/packages/FSharp.Compiler.Tools.4.0.0.1/tools/
-  - ci/download-asset.sh packages .
   - nuget restore JSIL.sln
   - nuget install NUnit.Runners -Version 2.6.4 -OutputDirectory packages 
   - nuget install FSharp.Compiler.Tools -Version 4.0.0.1 -OutputDirectory packages 
-  - ci/upload-asset.sh packages packages &
+
+before_script:
+  - mv /tmp/JSIL_Tests "/tmp/JSIL Tests" || true
 
 script:
   - xbuild /p:Configuration=Release /p:Platform=NoXna /v:m JSIL.sln
-  - ci/download-asset.sh compilecache /
   - mono ./packages/NUnit.Runners.2.6.4/tools/nunit-console.exe ./bin/Tests.DCE.dll ./bin/SimpleTests.dll ./bin/Tests.dll --run:$TestRun  --exclude:FailsOnMono\|\(FailsOnMonoWhenStubbed+Stubbed\)\|\(FailsOnMonoWhenStubbed+Translated\)
-  - ci/upload-asset.sh compilecache "/tmp/JSIL Tests"
 
+before_cache:
+  - mv "/tmp/JSIL Tests" /tmp/JSIL_Tests  
+  
+after_script:
+  - true && [ ! "$JSIL_STORAGE_TAG" ] && export BUILD_TAG=TRAVIS_${TRAVIS_JOB_NUMBER##*.} || export BUILD_TAG=$JSIL_STORAGE_TAG
+  - export CACHE_PATH=$BUILD_TAG/$TRAVIS_BUILD_NUMBER-$TRAVIS_COMMIT
+  - export UPLOAD_COMMAND="curl -H \"Authorization:OAuth $JSIL_STORAGE_KEY\" -L -T - \"https://webdav.yandex.com/JSIL/_\""  
```

Comment:
So I take it this replaces the service on my box with yandex webdav, and JSIL_STORAGE_KEY needs to be an oauth key?